docker/Dockerfile.prod

```diff
@@ -1,19 +1,42 @@
-FROM node:16.14.0 as base
+# First, install all dependencies, including devDependencies, to run the build process
+FROM node:16.14.0 as build
```

Comment:
This pattern is called a [multi-stage build](https://docs.docker.com/develop/develop-images/multistage-build/). At a high level, we have two distinct phases: `build` and `prod`.

The first stage, `build`, is defined here first. We install all dependencies, including `devDependencies` so `webpack` will work properly.

Then in the second stage, `prod`, we copy over the generated files, but only install the production dependencies to reduce the final image size.

This pattern reduced the image size from 2.51 GB to 1.43 GB (a 43% decrease!).